http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcClientConfigurationBuilder.java

```diff
@@ -176,6 +177,16 @@ public HttpClient call() {
         };
     }
 
+    public static OidcClientConfiguration buildWithoutUnsupportedAttributes(String unsupportedAttributesParam, InputStream is) {
+        OidcJsonConfiguration oidcJsonConfiguration = loadOidcJsonConfiguration(is);
+        try {
+            failIfUnsupportedAttribute(unsupportedAttributesParam, oidcJsonConfiguration);
```

Comment:
Just wondering if it's possible to detect the unsupported attribute somehow during `loadOidcJsonConfiguration` rather than after the `OidcJsonConfiguration` has been created?

Reply:
If we want it to be inside `loadOidcJsonConfig` to save parsing time, then don't think so, because loadOidcJsonConfig() uses `adapterConfig = mapper.readValue(is, OidcJsonConfiguration.class);`. And since scope is a jsonProperty defined in the OidcJsonConfiguration class, it would not fail. 
But if we want to do something like 
```
adapterConfig = mapper.readValue(is, OidcJsonConfiguration.class);
check for scope inside adapterConfig and fail if scope is not null
```
then yes, we can do that. 
But I am not sure if that simplifies the code any more. The reason why I added a new function is so that we still have a `build(InputStream is)` function. 
If we handle the scope check inside loadOidcJsonConfiguration, then we will instead need to add something like `loadOidcJsonConfigurationWithoutUnsuppportedAttributes(InputStream is, String unsupportedOidcAttributes)`
Please feel free to let me know if that is cleaner though. I am okay with changing it.